### PR TITLE
Support for CREATE OR REPLACE VIEW

### LINKED ISSUE #7606: sql: add support for CREATE OR REPLACE VIEW
PR: https://github.com/cockroachdb/cockroach/pull/47051

From release notes:
> Added support for the `CREATE OR REPLACE VIEW` command. [#47051][#47051] {% comment %}doc{% endcomment %}

### LINKED ISSUE #7746: Improve views: Create or Replace View
https://cockroachlabs.atlassian.net/browse/CRDB-742

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #7746. 
Fixes #7606. 

- Updated SQL diagram for `CREATE VIEW` page.
- Updated parameters on `CREATE VIEW` page.
- Updated examples on `CREATE VIEW` and Views pages.

### CODE REVIEW

File: v20.2/create-view.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ The user must have the `CREATE` [privilege](authorization.html#assign-privileges
 Parameter | Description
 ----------|------------
 `IF NOT EXISTS` |  Create a new view only if a view of the same name does not already exist. If one does exist, do not return an error.<br><br>Note that `IF NOT EXISTS` checks the view name only. It does not check if an existing view has the same columns as the new view.
+`OR REPLACE`  |  <span class="version-tag">New in v20.2:</span> If a view of the same name already exists, replace that view.<br><br>Note that `OR REPLACE` checks the view name only. It does not check if an existing view has the same columns as the new view.
```
Comment (rohany) [inline v20.2/create-view.md:24]: This is not quite correct. When we do replace view, we ensure that the new view has at least all of the same columns in the existing view as a prefix.

For example, if the old view had columns (a, b) the new view must have columns (a, b, insert, new, columns, here).

Comment (ericharmeling) [reply v20.2/create-view.md]: TFTR! I just updated the parameter description following your feedback.

File: v20.2/create-view.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ The user must have the `CREATE` [privilege](authorization.html#assign-privileges
 Parameter | Description
 ----------|------------
 `IF NOT EXISTS` |  Create a new view only if a view of the same name does not already exist. If one does exist, do not return an error.<br><br>Note that `IF NOT EXISTS` checks the view name only. It does not check if an existing view has the same columns as the new view.
+`OR REPLACE`  |  <span class="version-tag">New in v20.2:</span> If a view of the same name already exists, replace that view.<br><br>In order to replace an existing view, the new view must have the same columns as the existing view, or more. For example, if the existing view has columns `a, b`, the new view can have an additional column `c`, but must have columns `a, b`.
```
Comment (rohany) [inline v20.2/create-view.md:24]: We also should specify that the old columns must be a prefix of the old columns. We won't allow the view to have columns (b, a) in the case here.

Comment (ericharmeling) [reply v20.2/create-view.md]: Ah. Gotcha. I updated the description again.

**Pull request merged**